src/state-summary/state-card-display.html

```diff
@@ -40,7 +41,63 @@
   }
 
   computeStateDisplay(stateObj) {
-    return window.hassUtil.computeStateState(stateObj);
+    if (!stateObj._stateDisplay) {
+      const domain = window.hassUtil.computeDomain(stateObj);
+      if (domain === 'binary_sensor') {
+        var stateText = stateObj.state;
+        switch (stateObj.attributes.device_class) {
+          case 'moisture':
+            stateText = (stateObj.state === 'off') ? 'dry' : 'wet';
+            break;
+          case 'gas':
+          case 'motion':
+          case 'occupancy':
+          case 'smoke':
+          case 'sound':
+          case 'vibration':
+            stateText = (stateObj.state === 'off') ? 'clear' : 'detected';
+            break;
+          case 'opening':
+            stateText = (stateObj.state === 'off') ? 'closed' : 'open';
+            break;
+          case 'safety':
+            stateText = (stateObj.state === 'off') ? 'safe' : 'unsafe';
+            break;
+          default:
+        }
+        stateObj._stateDisplay = this.localize('state', stateText);
```

Comment:
we should have the keys be under binary_sensor domain

Reply:
Maybe also under device_class?

Reply:
Hmm, I like the idea of nesting under device_class as well. We could get rid of this branching entirely, and just look it up by device class.

The only concern I'd have is the case here where gas, motion, occupancy, smoke, sound and vibration all have the same results. Would we add duplicates in the schema? (Not opposed, since maybe alternates make sense in other translations.)